apps/src/applab/designMode.js

```diff
@@ -469,7 +469,38 @@ function makeDraggable (jqueryElements) {
         // resizable sets z-index to 90, which we don't want
         $(this).children().css('z-index', '');
       },
-      resize: function () {
+      resize: function (event, ui) {
+        // Wishing for a vector maths library...
+
+        // Customize motion according to current visualization scale.
+        var div = document.getElementById('divApplab');
+        var xScale = div.getBoundingClientRect().width / div.offsetWidth;
+        var yScale = div.getBoundingClientRect().height / div.offsetHeight;
+        var deltaWidth = ui.size.width - ui.originalSize.width;
+        var deltaHeight = ui.size.height - ui.originalSize.height;
+        var newWidth = ui.originalSize.width + (deltaWidth / xScale);
+        var newHeight = ui.originalSize.height + (deltaHeight / yScale);
```

Comment:
Sure.

Do we actually need to calculate xScale and yScale separately here?  We always lock aspect ratio, AFAIK.